app/services/candidate_submission.rb

```diff
@@ -0,0 +1,53 @@
+class CandidateSubmission
+  include Candidate
+
+  def initialize(session)
+    @session = session
+  end
+
+  def call
+    sign_up_candidate(select_candidate)
+  end
+
+private
+
+  def sign_up_candidate(body)
+    ApiClient.sign_up_teacher_training_adviser_candidate(body)
+  end
+
+  def select_candidate
+    case @session[:registration]["degree_options"]
+    when ReturningTeacher::DEGREE_OPTIONS[:returner]
+      filter_candidate(RETURNER)
+    when HaveADegree::DEGREE_OPTIONS[:yes]
+      filter_candidate(DEGREE)
+    when HaveADegree::DEGREE_OPTIONS[:studying]
+      filter_candidate(STUDYING)
+    when HaveADegree::DEGREE_OPTIONS[:equivalent]
+      filter_candidate(EQUIVALENT)
+    end
+  end
+
+  def filter_candidate(path)
+    data = @session[:registration].select { |key, _| path.include? key.to_sym }
+    merge_integers(data)
+  end
+
+  def merge_integers(body)
+    str_values = %w[
```

Comment:
I think this should be fixed before this PR is merged down. In my mind it should just be changing these attributes to type `:integer`:

https://github.com/DFE-Digital/get-teacher-training-adviser-service/blob/e74328ddae494c9580ec7d22e7bb4ef37807649e/app/models/what_degree_class.rb#L4

And then `to_s` the value here:

https://github.com/DFE-Digital/get-teacher-training-adviser-service/blob/master/app/validators/types_validator.rb#L5

Reply:
Not opposed to doing this, but it needs to be a sep PR, it touches a lot of files, and the verification process to check the correct data is getting posted is prolonged, I have opened a ticket.